clinica/utils/input_files.py

```diff
@@ -603,8 +603,14 @@ def bids_pet_nii(
     """
     from pathlib import Path
 
-    trc = "" if tracer is None else f"_trc-{tracer.value}"
-    rec = "" if reconstruction is None else f"_rec-{reconstruction.value}"
+    trc = ""
+    if tracer is not None:
+        tracer = Tracer(tracer)
+        trc = f"_trc-{tracer.value}"
+    rec = ""
+    if reconstruction is not None:
+        reconstruction = ReconstructionMethod(reconstruction)
+        rec = f"_rec-{reconstruction.value}"
     description = f"PET data"
```

Comment:
Maybe the variable `description` can be defined earlier as it is now so you can get rid of the redundancy and it would look like this : 

```
description = f"PET data"
trc = ""
if tracer is not None:
        tracer = Tracer(tracer)
        trc = f"_trc-{tracer.value}"
        description += f" with {tracer.value} tracer"
rec = ""
if reconstruction is not None:
        reconstruction = ReconstructionMethod(reconstruction)
        rec = f"_rec-{reconstruction.value}"
        description += f" and reconstruction method {reconstruction.value}"
return {...}
```

Reply:
Indeed ! Thanks for the suggestion @AliceJoubert ! 👍 
Implemented in [91adb0a](https://github.com/aramis-lab/clinica/pull/1207/commits/91adb0ada44ac7441bf220159b59285a66d2f9f4).